docs/book/common_code/noderef_controller.md

```diff
@@ -0,0 +1,8 @@
+# Node Controller
+
+The noderef controller populates a `Machine`'s `Status.NodeRef` field using the Kubernetes Node
+object associated with the Machine. This controller supports only machines linked to a cluster.
+
+Infrastructure providers can opt-in to use this controller by providing a KubeConfig as a Kubernetes Secret.
```

Comment:
I don't think I've ever seen kubeconfig capitalized as KubeConfig - maybe keep it lowercase, possibly in backticks?